src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -285,25 +285,41 @@ public function get(ProductId $productId): Product
         return $this->loadProduct($product);
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return ProductType
+     *
+     * @throws CoreException
+     * @throws Exception
+     * @throws ExceptionAlias
+     * @throws ProductConstraintException
+     * @throws ProductNotFoundException
+     */
     public function getProductType(ProductId $productId): ProductType
     {
-        $qb = $this->connection->createQueryBuilder();
-        $qb
-            ->addSelect('p.product_type')
+        $result = $this->connection->createQueryBuilder()
+            ->addSelect('p.product_type, p.cache_is_pack, p.is_virtual, p.cache_default_attribute')
             ->from($this->dbPrefix . 'product', 'p')
             ->where('p.id_product = :productId')
             ->setParameter('productId', $productId->getValue())
+            ->execute()
```

Comment:
IMO this change is not needed if the selected columns are not modified. Then you can 'just' have the real changes L317 to L322.

Reply:
The $qb variable is redundant, so I just removed the variable.